shared/editor/extensions/Mermaid.ts

```diff
@@ -23,74 +102,64 @@ function getNewState({
   doc: Node;
   name: string;
   pluginState: MermaidState;
-}) {
+}): MermaidState {
   const decorations: Decoration[] = [];
 
   // Find all blocks that represent Mermaid diagrams
-  const blocks: { node: Node; pos: number }[] = findBlockNodes(doc).filter(
+  const blocks = findBlockNodes(doc).filter(
     (item) =>
       item.node.type.name === name && item.node.attrs.language === "mermaidjs"
   );
 
+  let { initialized } = pluginState;
+  if (blocks.length > 0 && !initialized) {
+    void import("mermaid").then(({ default: mermaid }) => {
+      mermaid.initialize({
+        startOnLoad: true,
+        // TODO: Make dynamic based on the width of the editor or remove in
+        // the future if Mermaid is able to handle this automatically.
+        gantt: {
+          useWidth: 700,
+        },
+        theme: pluginState.isDark ? "dark" : "default",
+        fontFamily: "inherit",
+      });
+    });
+
+    initialized = true;
+  }
+
   blocks.forEach((block) => {
     const existingDecorations = pluginState.decorationSet.find(
       block.pos,
-      block.pos + block.node.nodeSize
+      block.pos + block.node.nodeSize,
+      (spec) => !!spec["diagramId"]
     );
 
-    // Attempt to find the existing diagramId from the decoration, or assign
-    // a new one if none exists yet.
-    const diagramId = existingDecorations[0]?.spec["diagramId"] ?? uuidv4();
+    const bestDecoration = last(
```

Comment:
Not sure I follow what's happening here, either we have an existing renderer/decoration with the correct ID – or not, no? Be worth a code comment.

Reply:
This issue happens when you have one diagram right under another. For example:
- Block 1: Span from 101 to 200
- Block 2: Span from 201 to 300

If you delete some text in Block 1, at this point the `block` value may span both Block 1 and 2, and we may accidentally get the Decoration for Block 2 when we update Block 1.

I have not fully investigated this behavior, however I found out that I can build a heuristic by finding the old Decoration node with most overlap over the recently edited block.